logstash-core/src/main/java/org/logstash/common/io/DeadLetterQueueWriter.java

```diff
@@ -252,15 +252,21 @@ private void innerWriteEntry(DLQEntry entry) throws IOException {
         executeAgeRetentionPolicy();
 
         if (currentQueueSize.longValue() + eventPayloadSize > maxQueueSize) {
-            if (storageType == QueueStorageType.DROP_NEWER) {
-                lastError = String.format("Cannot write event to DLQ(path: %s): reached maxQueueSize of %d", queuePath, maxQueueSize);
-                logger.error(lastError);
-                droppedEvents.add(1L);
-                return;
-            } else {
-                do {
-                    dropTailSegment();
-                } while (currentQueueSize.longValue() + eventPayloadSize > maxQueueSize);
+            // load currentQueueSize from filesystem because there could be a consumer
+            // that's already cleaning
+            this.currentQueueSize.set(computeQueueSize());
+
+            if (currentQueueSize.longValue() + eventPayloadSize > maxQueueSize) {
```

Comment:
Maybe extract this into a method for readability?

```suggestion
            if (isFreeQueueSpaceForEvent(eventPayloadSize)) {
```

Reply:
Already done in [PR 14324 ](https://github.com/elastic/logstash/pull/14324/files#diff-a0ee6ca8e72a830020520ea556f56e46ec1326e48593d9dfd1f252b70d3af45aR314) which I plan to merge before this, so adapt this to that, which generates a conflict.